src/ai/backend/agent/server.py

```diff
@@ -195,6 +200,33 @@ async def _inner(self_: AgentRPCServer, request: RPCMessage) -> Any:
         return _inner
 
 
+def _collect_metrics(observer: RPCMetricObserver) -> Callable:
+    def decorator(meth: Callable) -> Callable[[AgentRPCServer, RPCMessage], Any]:
+        @functools.wraps(meth)
+        async def _inner(self: AgentRPCServer, *args, **kwargs) -> Any:
+            start_time = time.perf_counter()
+            try:
+                res = await meth(self, *args, **kwargs)
+                duration = time.perf_counter() - start_time
+                observer.observe_rpc_request_success(
+                    method=meth.__name__,
+                    duration=duration,
+                )
+                return res
+            except Exception as e:
```

Comment:
How about handling `BaseException` here?

Reply:
Can you review again?